iothub/device/tests/MessageTests.cs

```diff
@@ -147,5 +147,29 @@ public void SettingMessageAsSecurityMessageTest()
             Assert.AreEqual(CommonConstants.SecurityMessageInterfaceId, msg.SystemProperties[MessageSystemPropertyNames.InterfaceId]);
             Assert.IsTrue(msg.IsSecurityMessage);
         }
+
+        [TestMethod]
+        public void CloneWithBodyTest()
+        {
+            var original = new Message(Encoding.UTF8.GetBytes("Original copy"));
+
+            original.Properties["test1"] = "test_v_1";
+            original.Properties["test2"] = "test_v_2";
+
+            original.ContentEncoding = "gzip";
+            original.UserId = "JohnDoe";
+
+            var clone = original.CloneWithBody(Encoding.UTF8.GetBytes("Cloned version"));
+
+            Assert.AreEqual("test_v_1", clone.Properties["test1"]);
+            Assert.AreEqual("test_v_2", clone.Properties["test2"]);
+
+            Assert.AreEqual("gzip", clone.ContentEncoding);
+            Assert.AreEqual("JohnDoe", clone.UserId);
+
+            var clonedContent = new StreamReader(clone.BodyStream, Encoding.UTF8).ReadToEnd();
```

Comment:
StreamReader is IDisposable, so put inside a using block

Reply:
I wonder if it is necessary here - everything is memory bound and gets dereferenced in a moment - dispose just unchains some dependencies in this case which would not make cleanup faster. On the other hand, it is easier to read without the using. Message instances are also disposable, yet the majority of the tests don't call dispose on them, even those requesting a body stream.

Reply:
It won't be disposed when it goes out of scope, right? That's the point of IDisposable objects, they have unmanaged backing resources that need to be explicitly disposed.

I know it is a test, but for a long test run, I'd prefer not to gain resource leaks. Also if we have fxcop validation in place, it prevents a code warning.

Reply:
if it has unmanaged resources, then it has a finalizer that will be called sooner or later and cleans up, so no resource leak would happen. It just cleans up later - but in this particular example, because it is all in memory, no difference I can see.
Anyway, I can easily put it into a using (or call Dispose)

Reply:
fixed and now dispose is called on disposable objects